File: be/src/main/java/ds/project/orino/common/response/exception/CustomException.java
package ds.project.orino.common.response.exception;

public class CustomException extends RuntimeException {

    private final ErrorCode errorCode;

    public CustomException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }

    public CustomException(ErrorCode errorCode, Throwable cause) {
        super(errorCode.getMessage(cause), cause);
        this.errorCode = errorCode;
    }

    public ErrorCode getErrorCode() {
        return errorCode;
    }
}

File: be/src/main/kotlin/ds/project/orino/common/response/exception/CustomException.kt
package ds.project.orino.common.response.exception


class CustomException : RuntimeException {
    val errorCode: ErrorCode

    constructor(errorCode: ErrorCode) : super(errorCode.message) {
        this.errorCode = errorCode
    }

    constructor(errorCode: ErrorCode, cause: Throwable) :
            super(errorCode.getMessage(cause), cause) {
        this.errorCode = errorCode
    }
}